javascript/ql/src/semmle/javascript/security/dataflow/UnsafeShellCommandConstructionCustomizations.qll

```diff
@@ -189,4 +189,25 @@ module UnsafeShellCommandConstruction {
       )
     }
   }
+
+  /**
+   * A check of the form `type x === "X"`, where x is  "number", or "boolean",
+   * which sanitized `x` in its "then" branch.
+   */
+  class TypeOfSanitizer extends TaintTracking::SanitizerGuardNode, DataFlow::ValueNode {
+    Expr x;
+    override EqualityTest astNode;
+
+    TypeOfSanitizer() {
+      exists(StringLiteral str, TypeofExpr typeof | astNode.hasOperands(str, typeof) |
+        str.getValue() = ["number", "boolean"] and // "undefined" is already handled in TaintTracking.qll
+        typeof.getOperand() = x
```

Comment:
A slightly more general and robust formulation is found here: https://github.com/github/codeql/blob/0c78fb293349ce964a72a7baadb2463166354008/javascript/ql/src/semmle/javascript/security/dataflow/UnsafeJQueryPluginCustomizations.qll#L137-L140